src/compiler.rs

```diff
@@ -331,6 +333,51 @@ impl Compiler {
         block.add(Op::Constant(value), self.line());
     }
 
+    fn grouping_or_tuple(&mut self, block: &mut Block) {
+        let block_length = block.ops.len();
+        let token_length = self.curr;
+        if self.try_tuple(block).is_err() {
+            block.ops.truncate(block_length);
+            self.curr = token_length;
+            self.grouping(block);
+        }
+    }
```

Comment:
Could be a macro seeing as we do this in two places now. `maybe_else!(self.try_truple, self.grouping, block)`

Reply:
I think this is a premature grouping. We might not always want to give `block` to all `try*` functions. There is a nice macro here, but that is a different PR.